src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -167,4 +193,62 @@ public JsonWebKey supplyKey(
 
         return keysByKid;
     }
+
+    private final class OAuthRealm
+    {
+        private static final int MAX_SCOPES = 48;
+
+        private final Map<String, Long> scopeBitsByName = new CopyOnWriteHashMap<>();
+
+        private final long realmId;
+        private final String realmName;
+
+        private long nextScopeBitShift;
+
+        private OAuthRealm(
+            String realmName,
+            long realmBitShift)
+        {
+            this.realmName = realmName;
+            this.realmId = 1L << realmBitShift << MAX_SCOPES;
+        }
+
+        private long assignScopeBit(
```

Comment:
Move this helper method after `resolve` and `lookup` which are the effective public API.